### PR TITLE
[test] Generalize runtime library searching

### PR DESCRIPTION
**Pull request by hamishknight**
Allow detecting the compiler runtime libraries for non-Darwin platforms, allowing us to test that programs relying on these libraries link and run successfully.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]
Seems like a nice cleanup. It would be nice to ensure that we also print the swift resource directory in addition to the clang one that you've cleaned up.

**Comment by hamishknight** [comment]
> It would be nice to ensure that we also print the swift resource directory in addition to the clang one that you've cleaned up.

I might be misunderstanding, but we should already be doing that:

https://github.com/swiftlang/swift/blob/251182cf5215607c9d0dddca87cbff214d062e55/test/lit.cfg#L379

**Comment by compnerd** [comment]
Ah, is just shuffled, so all good!

File: test/Profiler/coverage_inlined_counters_exec.swift [hunk 1]
```diff
@@ -21,6 +21,9 @@
 // REQUIRES: profile_runtime
 // REQUIRES: executable_test
 
+// FIXME: Currently fails on non-Darwin (https://github.com/swiftlang/swift/issues/75240)
+// REQUIRES: OS=macosx
```
Comment (etcwilde) [inline test/Profiler/coverage_inlined_counters_exec.swift:25]: Is this affected by the changes in this patch? It looks unrelated to me?

Comment (hamishknight) [reply test/Profiler/coverage_inlined_counters_exec.swift:25]: This is because we're now running this test on non-Darwin platforms, previously we wouldn't have picked up the `profile_runtime` feature

**Review by etcwilde — APPROVED** [review]
